packages/std/src/sections.rs

```diff
@@ -0,0 +1,64 @@
+/// A sections decoder for the special case of two elements
+#[allow(dead_code)] // used in Wasm and tests only
+pub fn decode_sections2(data: Vec<u8>) -> (Vec<u8>, Vec<u8>) {
+    let section2_len: usize = if data.len() >= 4 {
+        u32::from_be_bytes([
+            data[data.len() - 4],
+            data[data.len() - 3],
+            data[data.len() - 2],
+            data[data.len() - 1],
+        ]) as usize
+    } else {
+        panic!("Cannot read section2 length");
```

Comment:
Can you pull this into a helper function (it is used twice)?
The less code that needs to change to go from decode_sections2 -> decode_sections3 -> decode_sectionsN the better

Reply:
Also, why not return error instead of panic?
Not that anyone would handle them, but panics don't provide any useful message to the caller, errors help debugging if something did go wrong in the vm side.

Reply:
Because Rust's iterator interface does not support errors in `next()`

Reply:
:facepalm:  Good point